src/transformers/modeling_ctrl.py

```diff
@@ -462,7 +496,7 @@ def prepare_inputs_for_generation(self, input_ids, past, **kwargs):
         if past:
             input_ids = input_ids[:, -1].unsqueeze(-1)
 
-        return {"input_ids": input_ids, "past": past}
+        return {"input_ids": input_ids, "past": past, "use_cache": kwargs["use_cache"]}
```

Comment:
I think `use_cache` should be in the signature here.

Reply:
I did it because of `use_cache` comes after `attention_mask` in the signature. `attention_mask` is currently only implemented for encoder-decoder models because in "decoder-only" models like GPT2, batched input is not possible yet -> see: https://github.com/huggingface/transformers/issues/3021

We probably should allow batched input for GPT2 at some point though.